proto/internal/temporal/server/api/matchingservice/v1/service.proto

```diff
@@ -90,6 +90,18 @@ service MatchingService {
     // Apply a user data replication event.
     rpc ApplyTaskQueueUserDataReplicationEvent (ApplyTaskQueueUserDataReplicationEventRequest) returns (ApplyTaskQueueUserDataReplicationEventResponse) {}
 
+    // Gets all task queue names mapped to a given build ID
+    rpc GetBuildIdTaskQueueMapping (GetBuildIdTaskQueueMappingRequest) returns (GetBuildIdTaskQueueMappingResponse) {}
+
     // Force unloading a task queue. Used for testing only.
     rpc ForceUnloadTaskQueue (ForceUnloadTaskQueueRequest) returns (ForceUnloadTaskQueueResponse) {}
+
+    // Update task queue user data in owning node for all updates in namespace.
```

Comment:
this sounds too tempting to use when you want to update user data. can we make the comment very explicit that these two functions should only be called by the matching implementation itself and that external callers like frontend should use `UpdateWorkerBuildIdCompatibility` (or some other function for other fields)?

it might even be good if the name starts with `Internal`

Reply:
Can change to `Internal`